forward/rewriting.lua

```diff
@@ -99,12 +99,12 @@ function Rewriting:push()
    local i = assert(self.input.input, "input port not found")
    local o = assert(self.output.output, "output port not found")
    while not L.empty(i) do
-      self:process_packet(i, o)
+      local p = L.receive(i)
+      L.transmit(o, self:process_packet(p))
```

Comment:
What if `process_packet` decides to drop the packet instead of returning it?

Reply:
Right, missed that. Will fix later. I might also look into better ways of testing like making a custom source app.